### ISSUE TITLE
Made OpenSSL a framework requirement

### ISSUE THREAD

**Issue by fedemotta** [yiisoft/yii2#7661]
In advanced and basic apps requirements OpenSSL PHP extension is not being checked. However in the [Security documentation](http://www.yiiframework.com/doc-2.0/yii-base-security.html) says:
```
Note: this class requires 'OpenSSL' PHP extension for random key/string generation on Windows and for encryption/decryption on all platforms. For the highest security level PHP version >= 5.5.0 is recommended.
```

**Comment by fedemotta** [comment]
Pull request with the fix: #7663

**Comment by klimov-paul** [comment]
See https://github.com/yiisoft/yii2/pull/7239#discussion_r25566197
